devwilliamy/coverland
# John/preview order details v2 add tracking and billing

New Features:
Track Package Button / Payment Method

![Screenshot 2024-07-05 120355](https://github.com/devwilliamy/coverland/assets/171850605/603fbfda-4e11-4657-a384-b9c94f6c12ac)

https://app.clickup.com/t/8688z3p7v

https://www.figma.com/design/f0PiJaMbj3moDGXq45iAA1/Coverland-Website-Design?node-id=4599-58782&t=8KSJft3uQtluKfah-0

Testing Instructions:
-Go to an account order detail page
ie /profile/orders/3192
-test the tracking button (it should appear if an order has a tracking number associated)
 -on click it will take user to an outside tracking url
-test the different payment methods:
Options: Paypal, Cards (Visa, MasterCard, Discover, Amex, Google Pay, Apple Pay, Klarna
(you can use a test order inside Supabase DB and set the properties (ie card_brand, wallet_type, payment_gateway, payment_method)  if you dont want to create orders with these different payment methods)

Notes:
-1-click Link we don't store any billing info besides order.payment_method = "link"

![Screenshot 2024-07-05 115444](https://github.com/devwilliamy/coverland/assets/171850605/1d09618c-1482-4877-a9cb-5736892e16fb)
![Screenshot 2024-07-05 115912](https://github.com/devwilliamy/coverland/assets/171850605/33ee5384-f146-4d2a-bb84-206bce62cdbe)

-If Google Pay and Apple Pay are used, those are displayed, not card information:
![image](https://github.com/devwilliamy/coverland/assets/171850605/a28963fb-d570-4ca1-a89c-b58b36338fb2)

-Lastly a catch all if no payment method is found:

![Screenshot 2024-07-05 120134](https://github.com/devwilliamy/coverland/assets/171850605/2696827c-5e15-4535-83d0-b811765d8b96)
^ This doesn't really capture all scenario like if user paid with something like Apple Pay but the data failed to be retrieved. Maybe an error notice instead would be more appropriate? Like server error: data failed to load or try again later?
